pyproject.toml

```diff
@@ -35,7 +31,19 @@ classifiers = [
 dynamic = ["version"]
 
 [project.optional-dependencies]
-inference = ["onnxruntime==1.13.1"]
+all = [
+    "mpi4py>=3.1.4",
+    "netcdf4>=1.6.3",
+    "ruamel.yaml>=0.17.22",
+    "scikit-learn>=1.0.2",
+    "warp-lang>=0.6.0",
+    "tensorly>=0.8.1",
+    "vtk>=9.2.6",
+    "pyvista>=0.40.1",
+    "tensorly-torch>=0.4.0",
+    "torch-harmonics>=0.6.2",
```

Comment:
If I want to use SFNO and not some of the other models does this mean I will still have to install scikit-learn and warp-lang?

Reply:
Ideally you would run the model, it would error out the packages you need to install. Then you go again. Not sure theres a better solution I can think of

Reply:
Actually we could look at decomposing things into common groups more. Pandas does this... now if anyone would actually use this is a good question.

Reply:
Agree on Nick's point - I think we should have the script print out required packages to install. Also, the Launch recipe/example will have a `requiremtents.txt` that will install stuff specific for SFNO, so if someone is using the example, that should be a reference. I don't think we need to start decomposing/fragmenting too much just yet. The use case you presented @dallasfoster is a good one and we should continue to monitor such use cases before fragmenting. Basically, I am trying to avoid too much fragmentation of the `.toml` file for niche cases just yet.

Reply:
The only issue I have with @NickGeneva's solution, which I think goes to @ktangsali point, is that install by error is not a very user-friendly development design. If anything, I think something closer to 

```python
import_errors = []
try: 
    import torch_harmonics
except ImportError:
   import_errors.append("torch-harmonics")
...
assert len(import_errors) == 0, "Warning, need to install the following dependencies: " + str(import_errors)
```

Reply:
@dallasfoster is the suggestion here is to have such checks in each module?

Reply:
I agree this would be more user friendly. Some of the packages I see just error with an emphasis in the read me that you need to install all for everything to work but this still does lead to questions.

Similar ideas:
https://adamj.eu/tech/2021/12/29/python-type-hints-optional-imports/

At one point in time we had this type of check for some imports (like warp). I would probably suggest we address this on a case by case basis. I.e. if a user gets confused, we add in more detailed warnings. Otherwise, we just import error.

Reply:
Yeah, let's tackle this in future as further need arises. What do you think @dallasfoster ?

Reply:
Agreed, the present solution will suffice.